packages/template-retail-react-app/app/utils/product-utils.js

```diff
@@ -33,3 +33,21 @@ export const getDisplayVariationValues = (variationAttributes, values = {}) => {
     }, {})
     return returnVal
 }
+
+/**
+ * This function extract the promotional price from a product. If there are more than one price, the smallest price will be picked
```

Comment:
> If there are more than one price, the smallest price will be picked

I believe the assumption may be incorrect. We may need to combine multiple promotions to show the price.